src/utils/subscriberBreaches.ts

```diff
@@ -157,8 +161,28 @@ export async function getSubBreaches(
             (curBreach.dataClassesEffected[index][key] as number)++;
           }
         });
+
+        // Only mark data classes as resolved if they are resolved for all
+        // affected email addresses:
+        // We check if a data class is already resolved for the other email
+        // addresses. If that is the case, the respective data class are
+        // duplicated in `combinedResolvedDataClasses`. Those can be considered
+        // resolved and stay in the list of resolved data classes.
+        // Unique data classes will be filtered out as they haven’t been
+        // resolved for all affected email addresses.
+        const combinedResolvedDataClasses = [
+          ...curBreach.resolvedDataClasses,
+          ...resolvedDataClasses,
+        ];
+        const duplicateResolvedDataClasses = combinedResolvedDataClasses.filter(
+          (item, index) => combinedResolvedDataClasses.indexOf(item) !== index,
+        );
+        curBreach.resolvedDataClasses = duplicateResolvedDataClasses;
```

Comment:
The relevant fix here is this part: [Before](https://github.com/mozilla/blurts-server/pull/4212/files#diff-8d0a8d86a3d2ef197976f5229548d4001f281b3cfd77e4dc3014eb0f37e40ddfR164-R169) breach data classes were considered fixed if they have been fixed for one affected email address already.